packages/transaction-controller/src/utils/gas.ts

```diff
@@ -200,6 +202,66 @@ export function addGasBuffer(
   return maxHex;
 }
 
+/**
+ * Simulate the required gas for a batch of transactions using the simulation API.
+ *
+ * @param options - The options object.
+ * @param options.chainId - The chain ID of the transactions.
+ * @param options.from - The address of the sender.
+ * @param options.transactions - The array of transactions within a batch request.
+ * @returns An object containing the transactions with their gas limits and the total gas limit.
+ */
+export async function simulateGasBatch({
+  chainId,
+  from,
+  transactions,
+}: {
+  chainId: Hex;
+  from: Hex;
+  transactions: TransactionBatchSingleRequest[];
+}): Promise<{ transactions: TransactionBatchSingleRequest[]; gasLimit: Hex }> {
+  const response = await simulateTransactions(chainId, {
+    transactions: transactions.map((transaction) => ({
+      ...transaction.params,
+      from,
+    })),
+  });
+
+  if (
+    !response?.transactions ||
+    response.transactions.length !== transactions.length
+  ) {
+    throw new Error(
+      'Simulated gas response does not match the number of transactions',
+    );
+  }
+
+  let totalGasLimit = new BN(0);
+
+  const transactionsWithGasLimit = transactions.map((transaction, index) => {
```

Comment:
Minor, as we're converting a list to a single value, could we use `reduce`?

Reply:
Done.